tests/functional/spec/components/radio.spec.js

```diff
@@ -0,0 +1,109 @@
+                });
+
+          it('When I submit without choosing an option that I should be able to go on to the next page', function() {
+             return browser
+               .click(RadioNonMandatoryPage.submit())
+               .getUrl().should.eventually.contain(SummaryPage.pageName);
+            });
+
+         it('When I select an optional option field and not enter any text I should still be able to move on', function() {
+           return browser
+              .getUrl().should.eventually.contain(RadioNonMandatoryPage.pageName)
+              .click(RadioNonMandatoryPage.other())
+              .click(RadioNonMandatoryPage.submit())
+
+              .getText(SummaryPage.radioNonMandatoryAnswer()).should.eventually.contain('No answer provided');
+
+            });
+
+        it('When I select the other option then submit then return and remove other field text then it should be displayed correctly on the summary page', function() {
+         return browser
+          .click(RadioNonMandatoryPage.other())
+          .setValue(RadioNonMandatoryPage.otherText(), 'Hello')
+          .click(RadioNonMandatoryPage.submit())
+
+          .getText(SummaryPage.radioNonMandatoryAnswer()).should.eventually.contain('Hello')
+          .click(SummaryPage.previous())
+
+          .setValue(RadioNonMandatoryPage.otherText(), '')
+          .click(RadioNonMandatoryPage.submit())
+
+          .getText(SummaryPage.radioNonMandatoryAnswer()).should.eventually.contain('No answer provided');
+          });
+    });
+
+    describe('Mandatory', function() {
+      beforeEach('load the survey', function() {
+        return helpers.openQuestionnaire(radio_schema);
+            });
+
+      it('When I submit without choosing an option that I should get an error', function() {
+         return browser
+            .click(RadioMandatoryPage.submit())
+            .isVisible(RadioMandatoryPage.error()).should.eventually.be.true;
+        });
+
+      it('When I submit data it should be persisted and displayed on the summary page', function() {
+         return browser
+         .click(RadioMandatoryPage.bacon())
+         .click(RadioMandatoryPage.submit())
+
+         .click(RadioNonMandatoryPage.submit())
+
+         .getUrl().should.eventually.contain(SummaryPage.pageName)
+         .getText(SummaryPage.radioMandatoryAnswer()).should.eventually.contain('Bacon');
+        });
+
+
+      it('When I submit data and then go back and update an answer the content on the summary screen should be correct', function() {
+         return browser
+         .click(RadioMandatoryPage.eggs())
+         .click(RadioMandatoryPage.submit())
+
+         .click(RadioNonMandatoryPage.submit())
+
+         .getText(SummaryPage.radioMandatoryAnswer()).should.eventually.contain('Eggs')
+         .click(SummaryPage.previous())
+
+         .click(RadioNonMandatoryPage.submit())
+
+         .getText(SummaryPage.radioMandatoryAnswer()).should.eventually.contain('Eggs');
```

Comment:
This answer hasn't been changed?

Reply:
@jonnyshaw89 just discussed with @MIslam92 this proves persisted data, but he will add a test for changed answers